static/js/components/tiles/disaster_event_map_tile.tsx

```diff
@@ -0,0 +1,389 @@
+  }
+
+  /**
+   * Updates date info given an event type spec
+   */
+  function updateDateInfo(eventTypeSpec: Record<string, EventTypeSpec>): void {
+    const eventTypeDcids = Object.values(eventTypeSpec).flatMap(
+      (spec) => spec.eventTypeDcids
+    );
+    fetchDateList(eventTypeDcids)
+      .then((dateList) => {
+        setDateList(dateList);
+        if (!_.isEmpty(dateList)) {
+          setSelectedDate(dateList[0]);
+        }
+      })
+      .catch(() => {
+        setDateList([]);
+        setSelectedDate("");
+      });
+  }
+
+  /**
+   * Fetches the chart data needed for drawing the map
+   */
+  function fetchMapChartData(
+    selectedDate: string,
+    placeInfo: DisasterEventMapPlaceInfo,
+    eventTypeSpec: Record<string, EventTypeSpec>
+  ): void {
+    // Only re-fetch geojson data if place selection has changed
+    const geoJsonPromise =
+      mapChartData &&
+      !_.isEmpty(mapChartData.geoJson) &&
+      mapChartData.geoJson.properties.current_geo ===
+        placeInfo.selectedPlace.dcid
+        ? Promise.resolve(mapChartData.geoJson)
+        : fetchGeoJsonData(
+            placeInfo.selectedPlace.dcid,
+            placeInfo.enclosedPlaceType
+          );
+    const disasterEventDataPromise = fetchDisasterEventPoints(
+      Object.values(eventTypeSpec),
+      placeInfo.selectedPlace.dcid,
+      selectedDate
+    );
+    Promise.all([geoJsonPromise, disasterEventDataPromise])
+      .then(([geoJson, disasterEventData]) => {
+        setMapChartData({
+          geoJson,
+          disasterEventData,
+          sources: new Set(),
+        });
+      })
+      .catch(() => {
+        // TODO: add error message
+        setMapChartData(null);
+      });
+  }
+
+  /**
+   * Draws the disaster event map
+   */
+  function draw(
+    mapChartData: MapChartData,
+    placeInfo: DisasterEventMapPlaceInfo
+  ): void {
+    const width = svgContainerRef.current.offsetWidth;
+    const height = (width * 2) / 5;
+    const zoomParams = {
+      zoomInButtonId: ZOOM_IN_BUTTON_ID,
+      zoomOutButtonId: ZOOM_OUT_BUTTON_ID,
+    };
+    document.getElementById(props.id).innerHTML = "";
+    const isUsaPlace = isChildPlaceOf(
+      placeInfo.selectedPlace.dcid,
+      USA_PLACE_DCID,
+      placeInfo.parentPlaces
+    );
+    const canClickRegionCb = (placeDcid: string) => {
+      const newPlace = {
+        dcid: placeDcid,
+        name: placeDcid,
+        types: [placeInfo.enclosedPlaceType],
+      };
+      return canClickRegion(
+        newPlace,
+        placeInfo.selectedPlace,
+        placeInfo.parentPlaces,
+        europeanPlaces.current
+      );
+    };
+    const projection = getProjection(isUsaPlace, "", width, height);
+    drawD3Map(
+      props.id,
+      mapChartData.geoJson,
+      height,
+      width,
+      {} /* dataValues: no data values to show on the base map */,
+      "" /* units: no units to show */,
+      null /* colorScale: no color scale since no data shown on the base map */,
```

Comment:
For my own understanding, what's the reason for using /* */ type comments instead of // here?

Reply:
This was just because I had seen it done this way elsewhere in our code (https://github.com/datacommonsorg/website/blob/master/static/js/place/chart_block.tsx#L218), but either one should be fine!

Reply:
/* */ comments preferred for parameters :) https://google.github.io/styleguide/jsguide.html#formatting-param-name-comments